src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -656,17 +647,35 @@ Domain<3> BinaryCompactObject::create_domain() const {
 
 std::unordered_map<std::string,
                    std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
-BinaryCompactObject::functions_of_time() const {
+BinaryCompactObject::functions_of_time(
+    const std::unordered_map<std::string, double>& initial_expiration_times)
+    const {
   std::unordered_map<std::string,
                      std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
       result{};
   if (not enable_time_dependence_) {
     return result;
   }
 
-  const double initial_expiration_time =
-      initial_expiration_delta_t_ ? initial_time_ + *initial_expiration_delta_t_
-                                  : std::numeric_limits<double>::infinity();
+  // Get existing function of time names that are used for the maps and assign
+  // their initial expiration time to infinity (i.e. not expiring)
+  std::unordered_map<std::string, double> expiration_times{
+      {expansion_function_of_time_name_,
+       std::numeric_limits<double>::infinity()},
+      {rotation_about_z_axis_function_of_time_name_,
+       std::numeric_limits<double>::infinity()},
+      {size_map_function_of_time_names_[0],
+       std::numeric_limits<double>::infinity()},
+      {size_map_function_of_time_names_[1],
+       std::numeric_limits<double>::infinity()}};
+
+  // If we have control systems, overwrite these expiration times with the ones
+  // supplied by the control system
+  for (auto& [name, expr_time] : initial_expiration_times) {
+    if (expiration_times.count(name) == 1) {
```

Comment:
just delete this `if`? Inconsistencies should be caught by the checks in the initialization right